dashboard/app/models/experiment.rb

```diff
@@ -25,56 +25,41 @@
 #  index_experiments_on_section_id            (section_id)
 #
 
-MAX_CACHE_AGE = 1.minute
+MAX_CACHE_AGE = Rails.application.config.experiment_cache_time_seconds.seconds
 
 class Experiment < ApplicationRecord
   belongs_to :script, optional: true
 
   validates :name, presence: true
-  after_create {Experiment.update_cache}
+  after_save {Experiment.update_cache}
+  after_destroy {Experiment.update_cache}
 
   @@experiments = nil
 
-  def self.should_cache?
-    Script.should_cache?
-  end
-
-  def self.get_all_enabled(user: nil, section: nil, script: nil)
-    if Experiment.should_cache?
-      Experiment.get_all_enabled_cached(user: user, section: section, script: script)
-    else
-      Experiment.get_all_enabled_uncached(user: user, section: section, script: script)
-    end
-  end
-
-  def self.get_all_enabled_uncached(user: nil, section: nil, script: nil)
-    experiments = Experiment.descendants.map do |experiment_type|
-      experiment_type.get_enabled(user: user, section: section, script: script)
-    end
-    experiments = experiments.reduce(Experiment.none) do |e1, e2|
-      e1.or(e2)
-    end
-    now = DateTime.now
-    experiments.where('start_at IS NULL OR start_at < ?', now).
-      where('end_at IS NULL OR end_at > ?', now).
-      where('script_id IS NULL OR script_id = ?', script.try(:id))
-  end
-
-  def self.get_all_enabled_cached(user: nil, section: nil, script: nil)
+  def self.get_all_enabled(user: nil, section: nil, script: nil, experiment_name: nil)
     if @@experiments.nil? || @@experiments_loaded < DateTime.now - MAX_CACHE_AGE
-      Experiment.update_cache
+      update_cache
     end
     now = DateTime.now
     @@experiments.select do |experiment|
       experiment.enabled?(user: user, section: section) &&
         (experiment.start_at.nil? || experiment.start_at < now) &&
         (experiment.end_at.nil? || experiment.end_at > now) &&
-        (experiment.script_id.nil? || experiment.script_id == script.try(:id))
+        (experiment.script_id.nil? || experiment.script_id == script.try(:id)) &&
+        (experiment_name.nil? || experiment.name == experiment_name)
     end
   end
 
+  def self.enabled?(user: nil, section: nil, script: nil, experiment_name: nil)
+    get_all_enabled(
+      user: user,
+      section: section,
+      script: script,
+      experiment_name: experiment_name
+    ).any?
+  end
+
   def self.update_cache
-    return unless Experiment.should_cache?
     @@experiments = Experiment.all.to_a
```

Comment:
Do we need to be worried about the size of this? In particular, I could imagine this getting obscenely huge down the road. Is it sufficient to load the set of experiments that are active?